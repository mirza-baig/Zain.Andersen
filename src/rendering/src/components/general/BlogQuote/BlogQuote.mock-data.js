const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'BlogQuote',
  },
  fields: {
    quoteName: {
      value: 'Saad Patel',
    },
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    body: {
      value:
        '<h3>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</h3>',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
