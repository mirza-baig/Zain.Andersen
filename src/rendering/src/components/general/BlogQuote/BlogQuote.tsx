// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { Text, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import { ComponentProps } from 'lib/types/component-props';
import { getBreakpoint, useCurrentScreenType } from 'lib/utils/get-screen-type';

const QuoteIcon = () => {
  const { currentScreenWidth } = useCurrentScreenType();

  return (
    <svg
      width={currentScreenWidth <= getBreakpoint('md') ? 41 : 54}
      height={currentScreenWidth <= getBreakpoint('md') ? 26 : 34}
      viewBox="0 0 54 35"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M53.88 0.479994L43.08 0L24.84 33.48L40.56 34.32L53.88 0.479994ZM28.92 0.479994L18.12 0L0 33.48L15.72 34.32L28.92 0.479994Z"
        fill="#F26924"
      />
    </svg>
  );
};

export type BlogQuoteProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.BlogQuote.BlogQuote & ComponentProps;
const BlogQuote = (props: BlogQuoteProps) => {
  const { fields } = props;

  return (
    <Component variant="lg" dataComponent="general/blogquote" {...props}>
      <div className="col-span-12">
        <div className="flex justify-start">
          <div className="mr-xs md:mr-m">
            <QuoteIcon />
          </div>
          <div className="font-sans text-sm-xs font-medium md:text-xs">
            <RichTextWrapper field={fields?.body} refer="font-heavy text-sm-s md:!text-s mb-xxs" />
            <Text
              tag="p"
              field={{
                value: fields?.quoteName.value,
              }}
            />
          </div>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<BlogQuoteProps>(BlogQuote);
