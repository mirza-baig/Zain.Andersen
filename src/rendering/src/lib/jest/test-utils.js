// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
