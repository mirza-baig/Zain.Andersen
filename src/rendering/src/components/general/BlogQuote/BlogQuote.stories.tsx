// Global
import { Story, Meta } from '@storybook/react';
// Local
import BlogQuote, { BlogQuoteProps } from './BlogQuote';
import defaultData from './BlogQuote.mock-data';

export default {
  title: 'General/BlogQuote',
  component: BlogQuote,
} as Meta;
const Template: Story<BlogQuoteProps> = (props) => <BlogQuote {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
