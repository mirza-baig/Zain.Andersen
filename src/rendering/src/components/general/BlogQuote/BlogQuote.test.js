// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import BlogQuote from './BlogQuote';
import defaultData from './BlogQuote.mock-data';

it('renders correctly', async () => {
  const component = snapshot(BlogQuote, { componentProps: defaultData });
  await hasDataComponent(component, 'general/blogquote');
});
