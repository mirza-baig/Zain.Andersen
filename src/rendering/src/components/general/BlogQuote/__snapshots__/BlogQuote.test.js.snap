// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/blogquote"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <div
          class="flex justify-start"
        >
          <div
            class="mr-xs md:mr-m"
          >
            <svg
              fill="none"
              height="34"
              viewBox="0 0 54 35"
              width="54"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M53.88 0.479994L43.08 0L24.84 33.48L40.56 34.32L53.88 0.479994ZM28.92 0.479994L18.12 0L0 33.48L15.72 34.32L28.92 0.479994Z"
                fill="#F26924"
              />
            </svg>
          </div>
          <div
            class="font-sans text-sm-xs font-medium md:text-xs"
          >
            <div
              class="font-heavy text-sm-s md:!text-s mb-xxs"
            >
              <div>
                <h3>
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                </h3>
              </div>
            </div>
            <p>
              Saad Patel
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
